src/utils/commands.ts

```diff
@@ -0,0 +1,71 @@
+import { Message, Permissions } from 'discord.js';
+import Commando, { CommandoMessage, CommandoClient, CommandInfo, ArgumentCollectorResult } from 'discord.js-commando';
+import logger, { logError } from '../components/logger';
+
+export abstract class BaseCommand extends Commando.Command {
+  constructor(client: CommandoClient, info: CommandInfo) {
+    super(client, info);
+  }
+
+  run(
```

Comment:
Can you explain the difference between `run()` and `onRun()` ? 
Is `onRun` a just a function header?

Reply:
Did Github just repeat my comment lol?

Reply:
`run` is an abstract method in Commando's command class that we have to implement, which is what Commando will run when a command is triggered. `onRun` is an abstract method in our own `BaseCommand` class, where any commands we implement that inherit/extend from `BaseCommand` will have to implement. In `BaseCommand`'s `run` method, you can see that we first log the command, then call `onRun`, so that's what happens when a command is triggeered.

Reply:
In OOP, an abstract method is kinda like a header like you said, where it doesn't contain an implementation. Any subclasses that inherit that class will need to implement any abstract methods it has.

Reply:
Ah I see I see - Thanks Charles!